compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -889,7 +889,12 @@ trait Checking {
         if (decl.matches(other) && !javaFieldMethodPair) {
           def doubleDefError(decl: Symbol, other: Symbol): Unit =
             if (!decl.info.isErroneous && !other.info.isErroneous)
-              report.error(DoubleDefinition(decl, other, cls), decl.srcPos)
+              if decl.owner.is(Enum, butNot=JavaDefined|Case) && decl.span.isSynthetic && (
+                decl.name == nme.ordinal || decl.name == nme.enumLabel)
+              then
+                report.error(EnumGettersRedefinition(decl), other.srcPos)
```

Comment:
By now you have probably guessed that I hate special cases; they are the source of almost all evil in a compiler. So, do we really need this?

Reply:
Agreed this probably does not need to be there, you also get Double Definition error if you try to define `def values: Array[MyEnum]`, I've just checked and java does nothing special, just override final method error

Reply:
this is now removed